devito/at_controller.py

```diff
@@ -91,71 +95,69 @@ def auto_tune_blocks(self, minimum=5, maximum=20):
         at_nt = 3
         self.op.propagator.nt = at_nt
         self.op.propagator.profile = True
-        self.op.propagator.cache_blocking = [0 if dim else None
-                                             for dim in self.blocked_dims]
-        # want to run function at least once to make sure block_sizes are not
-        # overwritten
-        self.get_execution_time()
-        logger.info("Starting auto tuning of block sizes using brute force")
-
-        times = []  # list where times and block sizes will be kept
+
+        info_at("Start. Mode: brute force")
+
         block_list = set()  # used to make sure we do not test the same block sizes
-        blocks = list(self.op.propagator.block_sizes)
+        mask = [i if i else None for i in self.blocked_dims]
+        block = [None for i in mask]
 
         for x in range(minimum, maximum):
-            blocks[0] = x if blocks[0] else None
+            block[0] = mask[0] and x
 
-            if len(blocks) > 1:
+            if len(block) > 1:
                 for y in range(minimum, maximum):
```

Comment:
ok, I'm changing ``for y in range(minimum, maximum):`` into ``for y in range(minimum, x + 1):``

as for the non-squared block sizes: I agree they might be better -- but by how much ? is the gain ever significant ? I'm happy to relax the filtering once we have compelling reasons to do so